lib/trino-orc/src/main/java/io/trino/orc/OrcReader.java

```diff
@@ -463,8 +463,8 @@ public static ProjectedLayout createProjectedLayout(OrcColumn root, List<List<St
 
             Map<String, List<List<String>>> dereferencesByField = dereferences.stream().collect(
                     Collectors.groupingBy(
-                            sequence -> sequence.get(0),
-                            mapping(sequence -> sequence.subList(1, sequence.size()), toList())));
+                            sequence -> sequence.get(0).toLowerCase(ENGLISH),
+                            mapping(sequence -> sequence.subList(1, sequence.size()).stream().map(String::toLowerCase).collect(toList()), toList())));
```

Comment:
replace `.map(String::toLowerCase)` with lambda using `.toLowerCase(ENGLISH)` 

also, since you lowercase everything, it's actually better to pull this before the groupBy

```
           Map<String, List<List<String>>> dereferencesByField = dereferences.stream()
                    .map(list -> list.stream()
                            .map(dereference -> dereference.toLowerCase(ENGLISH))
                            .collect(toImmutableList()))
                    .collect(groupingBy(
                            sequence -> sequence.get(0),
                            mapping(sequence -> sequence.subList(1, sequence.size()), toList())));
```

Reply:
I pulled the `toLowerCase` out ahead of the `groupBy`, it still looks a little odd to me so let me know if its what you are looking for.